### DISCUSSION TITLE
Task 4 - CARD TRICK

### DISCUSSION THREAD

**Discussion by ThomasPineda** [rramosp/ai4eng.v1#42] [02 - PYTHON]
![Captura de pantalla 1943-04-10 a la(s) 11 23 03 a  m](https://user-images.githubusercontent.com/86691796/124157769-c0b80b00-da5e-11eb-9fde-80f25f8b1fdb.png)
![Captura de pantalla 1943-04-10 a la(s) 11 23 34 a  m](https://user-images.githubusercontent.com/86691796/124157802-cb72a000-da5e-11eb-88d9-2ab004b4bfa3.png)
![Captura de pantalla 1943-04-10 a la(s) 11 23 51 a  m](https://user-images.githubusercontent.com/86691796/124157842-d6c5cb80-da5e-11eb-9db6-28c5081ffe1c.png)

Creo que hago el codigo como debe ser, las posiciones lo hago de acuerdo a la formula que se da en el ejercicio, cuando hago el chequeo del código me sale bien, pero cuando ya voy a calificar mi tarea aparece que la formula no saca un numero entero y distinto al de la posición que se esperaría y no entiendo por qué

**Comment by Felipe-RA** [comment]
Saludos ThomasPineda

![image](https://user-images.githubusercontent.com/49454068/124313905-2aa3e380-db37-11eb-8c45-6fbb8d53b6cd.png)

Recuerda que este ejercicio de las cartas hace uso de las [propiedades de la división entera](https://rua.ua.es/dspace/bitstream/10045/26436/1/Matematica%20Discreta.pdf)

Es por esto que no es correcto aplicar la división real en un problema del cual requerimos obtener resultados **discretos**, para esto tenemos la división entera, la cual si que arroja resultados discretos.

Te recomiendo este pequeño extracto sobre la **división entera y la aritmética modular**, de la mano de Khan Academy https://es.khanacademy.org/computing/computer-science/cryptography/modarithmetic/a/what-is-modular-arithmetic

Hasta pronto!

**Reply by Moorea-AI** [reply]
Muchas gracias por esta respuesta, media hora después de estar probando de todo me dicuenta de que el error era la división (solo porque vine aqui a ver si alguien más habia tenido mi problema jajaajajaj)    :)
